pkg/csi_driver/multishare_ops_manager.go

```diff
@@ -337,17 +337,71 @@ func (m *MultishareOpsManager) instanceNeedsExpand(ctx context.Context, share *f
 		sumShareBytes = sumShareBytes + s.CapacityBytes
 	}
 
-	var remainingBytes int64
-	remainingBytes = share.Parent.CapacityBytes - sumShareBytes
-	if remainingBytes < share.CapacityBytes {
-		alignBytes := util.AlignBytes(share.CapacityBytes+sumShareBytes, util.GbToBytes(share.Parent.CapacityStepSizeGb))
+	remainingBytes := share.Parent.CapacityBytes - sumShareBytes
+	if remainingBytes < capacityNeeded {
+		alignBytes := util.AlignBytes(capacityNeeded+sumShareBytes, util.GbToBytes(share.Parent.CapacityStepSizeGb))
 		targetBytes := util.Min(alignBytes, util.MaxMultishareInstanceSizeBytes)
 		return true, targetBytes, nil
 	}
 	return false, 0, nil
 }
 
-func (m *MultishareOpsManager) checkAndStartShareDeleteWorkflow(ctx context.Context, instanceScPrefix string, share *file.Share) (*Workflow, error) {
+func (m *MultishareOpsManager) expandInstanceAndStartWorkflow(ctx context.Context, share *file.Share, reqBytes int64) (*Workflow, error) {
```

Comment:
checkAndStartInstanceExpandWorkflow is a better name

Reply:
done